libs/langchain/langchain/memory/chat_memory.py

```diff
@@ -16,18 +16,21 @@ class BaseChatMemory(BaseMemory, ABC):
     return_messages: bool = False
 
     def _get_input_output(
-        self, inputs: Dict[str, Any], outputs: Dict[str, str]
+        self, inputs: Dict[str, Any], outputs: Dict[str, Any]
     ) -> Tuple[str, str]:
         if self.input_key is None:
             prompt_input_key = get_prompt_input_key(inputs, self.memory_variables)
         else:
             prompt_input_key = self.input_key
+
         if self.output_key is None:
-            if len(outputs) != 1:
-                raise ValueError(f"One output key expected, got {outputs.keys()}")
-            output_key = list(outputs.keys())[0]
+            if 'answer' in outputs:
```

Comment:
could you explain the use case/issue you're seeing? don't think this is very generalizable

Reply:
Hi, so I could not return_source_documents in a ConversatinoRetreivalChain, so I wrote a temporary thing to do it, however, the more general approach would be better, can you fix it?